course5/homework04/homework04.cc

```diff
@@ -0,0 +1,122 @@
+#include <iostream>
+#include <cstdio>
+#include <cstdlib>
+#include <stdexcept>
+#include <string>
+
+/***
+Define a “Stream” class, that allows for easy printing of values to a file
+- Let the class manage the file: ie open the file in constructor and close it in destructor.
+- Overload operator<< for the class, so that it can be used to print some types.
+- Define a global variable of the “Stream” type, call it, say, “Cout”, that will write to stdout.
+***/
+
+class Stream
+{
+public:
+	// Constructors
+	Stream() // default
+	{
+		throw std::runtime_error("Stream must be constructed with a filename: Stream('filename.txt')");
+	}
+	Stream(const Stream&) // copy
+	{
+		throw std::runtime_error("Stream object cannot be copied");
+	}
+	Stream(Stream&& src) // move
+	{
+		fname = src.fname;
+		fp = src.fp;
+	}
+	Stream(const char* fname)
+	{
+		this->fname = fname;
+		this->fp = std::fopen(fname, "a");
+	}
+	Stream(std::FILE *)
+	{
+		this->fp = stdout;
+	}
+
+	// Destructor
+	~Stream()
+	{
+		std::fclose(this->fp);
+	}
+
+	// copy assignment operator
+	Stream& operator=(const Stream& src)
+	{
+		fname = src.fname;
+		fp = src.fp;
+		return *this;
+	}
+
+	// move assignment operator
+	Stream& operator=(Stream&& src)
+	{
+		fname = src.fname;
+		fp = src.fp;
+		return *this;
+	}
+
+	// overloaded << operator
+	Stream& operator << (int input)
+	{
+		if (input == EOF)
```

Comment:
Maybe I'm using it wrong then? Because I'm passing it in as an input when I want a newline - I don't understand what you mean by "return value" here.